### ISSUE TITLE
Validation breaks with collection rules

### ISSUE THREAD

**Issue by martinhla** [victorgarciaesgi/regle#22]
Hi, I'm encountering an error while validating collections with the following configuration:

```
const { r$ } = useRegle(data, {
  name: { required },
  levels: {
    minLength: minLength(1),
    maxLength: maxLength(3),
    $each: () => ({
      name: { required },
      localizations: {
        $each: () => ({
          name: { required },
        }),
      },
    }),
  },
});
```

When I attempt to add a level: `{ name: 'a', localizations: [{ name: 'a' }] }`, I receive the following error:
**Uncaught (in promise) TypeError: Cannot read properties of undefined (reading '0')**

**Reproduction:** 
https://stackblitz.com/edit/vitejs-vite-wa6v5z?file=src%2FApp.vue just click on add level button and you will see the error

**Comment by victorgarciaesgi** [comment]
Thanks for the report! I must have missed something with nested collections callbacks. I'll take a look

**Comment by victorgarciaesgi** [comment]
Luckily just a mistake on my part on an undefined check, easy fix. Will be pushed in the next update 👍

**Comment by martinhla** [comment]
thanks for fix. i will try that after update 👍

**Comment by victorgarciaesgi** [comment]
Fixed in `0.2.11`
